url5c7c5a5b36_taczge_rdf_store_tgz-pJ8-experiment_SATSolverExperimentJ8/src/query/primitive/PrimitiveQueryFactory.java

```diff
@@ -0,0 +1,86 @@
+package query.primitive;
+
+import query.PrimitiveQuery;
+import query.Token;
+import query.Variable;
+import core.Resource;
+
+public class PrimitiveQueryFactory {
+
+	private PrimitiveQueryFactory() {}
+	
+	public static PrimitiveQuery create(Token s, Token p, Token o) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.